vsintegration/src/FSharp.Editor/Common/CommonHelpers.fs

```diff
@@ -341,6 +341,62 @@ module internal CommonHelpers =
         | -1 | 0 -> span
         | index -> TextSpan(span.Start + index + 1, text.Length - index - 1)
 
+    let isValidNameForSymbol (lexerSymbolKind: LexerSymbolKind, symbol: FSharpSymbol, name: string) : bool =
+        let doubleBackTickDelimiter = "``"
+        
+        let isDoubleBacktickIdent (s: string) =
+            let doubledDelimiter = 2 * doubleBackTickDelimiter.Length
+            if s.StartsWith(doubleBackTickDelimiter) && s.EndsWith(doubleBackTickDelimiter) && s.Length > doubledDelimiter then
+                let inner = s.Substring(doubleBackTickDelimiter.Length, s.Length - doubledDelimiter)
+                not (inner.Contains(doubleBackTickDelimiter))
+            else false
+        
+        let isIdentifier (ident: string) =
+            if isDoubleBacktickIdent ident then
+                true
+            else
+                ident 
+                |> Seq.mapi (fun i c -> i, c)
+                |> Seq.forall (fun (i, c) -> 
+                     if i = 0 then PrettyNaming.IsIdentifierFirstCharacter c 
+                     else PrettyNaming.IsIdentifierPartCharacter c) 
+        
+        let isFixableIdentifier (s: string) = 
+            not (String.IsNullOrEmpty s) && Lexhelp.Keywords.NormalizeIdentifierBackticks s |> isIdentifier
+        
+        let forbiddenChars = ["."; "+"; "$"; "&"; "["; "]"; "/"; "\\"; "*"; "\""]
+        
+        let isTypeNameIdent (s: string) =
+            not (String.IsNullOrEmpty s) &&
+            forbiddenChars |> Seq.forall (fun c -> not (s.Contains c)) &&
```

Comment:
`String.IndexOfAny` would probably be a bit faster but personally I love how this one rolls:
`not (Seq.exists s.Contains forbiddenChars)`
I don't think it would be idiomatic, though.
Anyway, what a sweet, sweet language F# is. It makes me grin with joy anytime I read or write it :)